### DISCUSSION TITLE
Does @xstate/test support xstate V5?

### DISCUSSION THREAD

**Discussion by iliasbhal** [statelyai/xstate#4761] [Q&A] [answered]
👋  Hey Community!

I was starting a fresh project, so naturally I'm using xstate and nonetheless the V5.
I didn't have luck generating tests using @xstate/test. 
I started by recreating the example in the docs but it didn't work 🥲

```ts
 const toggleMachine = x.createMachine({
    id: "toggle",
    initial: "inactive",
    states: {
      inactive: {
        on: {
          TOGGLE: "active",
        },
        meta: {
          test: async () => {
            console.log("TEST INACTIVE");
          },
        },
      },
      active: {
        on: {
          TOGGLE: "inactive",
        },
        meta: {
          test: async () => {
            console.log("TEST ACTIVE");
          },
        },
      },
    },
  });

  const toggleModel = xt.createModel<{}>(toggleMachine).withEvents({
    TOGGLE: {
      exec: async () => {
        // await page.click('input');
      },
    },
  });

  const testPlans = toggleModel.getShortestPathPlans();
  testPlans.forEach((plan) => {
    describe(plan.description, () => {
      plan.paths.forEach((path) => {
        it(path.description, async () => {
          await path.test({});
        });
      });
    });
  });
```

Results in:
<img width="1201" alt="Screenshot 2024-02-24 at 12 45 42 AM" src="https://github.com/statelyai/xstate/assets/9442803/0a515dd7-0058-48a5-a7a3-401c268d1dc1">

**Comment by davidkpiano** [comment, marked as answer]
Hey! Have you tried `@xstate/test@beta`?

**Reply by iliasbhal** [reply]
Thank for the hint :) 
Works like a charm 🤘

**Reply by dillonrowan** [reply]
I'm trying to import createModel into my nextjs project via `import { createModel } from '@xstate/test'`. I am getting `Module not found: Can't resolve '@xstate/test'` What would be the correct syntax to import this?
`import { createModel } from "@xstate/test@beta";` is also giving an error

**Reply by davidkpiano** [reply]
@dillonrowan You want `createTestModel(…)`

**Reply by dillonrowan** [reply]
Thank you for response. I can not find how `createTestModel()` is used in the xstate v5 website. Do you maybe have an example on how one would test their xstate v5 state machine with `createTestModel()`?

**Reply by davidkpiano** [reply]
Sorry for the lack of docs - will try to get to that soon.

But you can always look at the tests to see how it's used: https://github.com/statelyai/xstate/tree/main/packages/xstate-test/test/index.test.ts#L61

**Reply by dillonrowan** [reply]
Is it private? I am not able to get to this link. Getting not found 404 error.

**Reply by davidkpiano** [reply]
Sorry, there was a typo. Try the same link again?

**Reply by wencyen** [reply]
Hi David,

The link works! Thanks. Do you and the stately team have a timeline for releasing a stable version for the `@xstate/test` that is compatible with v5 of xstate? I too am a new comer to the xstate library but have been really enjoying modeling my application state using fsm.

**Reply by davidkpiano** [reply]
I will try to get it to the finish line next week 🤞

**Reply by cbishop-fv** [reply]
> I will try to get it to the finish line next week 🤞

Looking at the changelog I belive this is still in beta? We are also using and loving xstate5 and would love to test with a stable version. Thanks!

**Reply by davidkpiano** [reply]
> Looking at the changelog I belive this is still in beta? We are also using and loving xstate5 and would love to test with a stable version. Thanks!

@cbishop-fv The plan is to release a major version of `@xstate/graph`, which includes everything from `@xstate/test`. It's being consolidated into one package since they are closely related, and I want to emphasize that `@xstate/test` is not for testing XState machines, it's for _using_ XState machines to test anything.

**Reply by zara-khaliq** [reply]
The link posted above is private. Are there any example docs to see how I can use xstate/test with version 5. Having docs using react native and react resting library would be great!

**Reply by zara-khaliq** [reply]
I created a test.tsx file with the following code however I'm gaving problems running the test. jest doesn't seem to recognize the xstate/test import. Is this how you're meant to test the machine. 
```
import {render, screen} from '@testing-library/react-native';
import {createModel} from '@xstate/test';

import {LoginScreen} from '../../LoginScreen';
import {LoginScreenMachine} from '.';

import {WahedTheme} from '~/WahedTheme';

const LoginMachineModel = createModel<any>(LoginScreenMachine);

const testPlans = LoginMachineModel.getShortestPaths();
describe('Component', () => {
  testPlans.forEach((path) => {
    it.concurrent(path.description, async () => {
      render(
        <WahedTheme>
          <LoginScreen />
        </WahedTheme>
      );
      await path.test({
        states: {
          fetchingPreviousLoginDetails: async () => {
            // not sure how UI will look like in this state
          },
          idle: async () => {
            await screen.findByText('Login');
          },
          loggingIn: async () => {
            await screen.getByRole('progressbar');
          },
          onboarding: async () => {
            // should expect this to not there
            await screen.findByText('Login');
          }
        },
        events: {
          PRESSED_LOG_IN: async () => {
            fireEvent.press(await screen.findByText('Login'));
          },
          LEGACY_USA_USER_LOGGED_IN: async () => {
            fireEvent.press(await screen.findByText('Login'));
          },
          CHANGE_EMAIL: async () => {
            fireEvent.changeText(
              await screen.findByTestId('input-email'),
              ' [email protected]'
            );
          },
          CHANGE_PASSWORD: async () => {
            fireEvent.changeText(
              await screen.findByTestId('input-password'),
              'password'
            );
          }
        }
      });
    });
  });
});
```

**Reply by pkelly-rh** [reply]
[zara-khaliq](https://github.com/zara-khaliq)
The api is changed. use createTestModel instead of createModel
something like this:
import { createTestModel } from "@xstate/test";
